knowledge-base/grid-customize-hierarchy-expand-column-blazor-grid.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Customize Hierarchy Expand Column in Blazor Grid
+description: Learn how to customize the hierarchy expand/collapse column in Telerik Blazor Grid, including changing its position, locking it, and setting its width and title.
+type: how-to
+page_title: How to Customize the Hierarchy Expand/Collapse Column in Telerik Blazor Grid
+slug: grid-kb-customize-hierarchy-expand-column-blazor-grid
+tags: grid, blazor, hierarchy, expand, collapse, column, customize
+res_type: kb
+ticketid: 1658470
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB article answers the following questions:
+
+- Can I change the position of the hierarchy expand/collapse column?
+- Is it possible to lock/freeze the expand column for a hierarchical grid?
+- Can I add a title to the hierarchy expand/collapse column?
+- How can I change the width of the expand column in a hierarchical Grid?
+
+## Solution
+
+By default, the hierarchy expand/collapse column in the Telerik UI for Blazor [Grid]({%slug grid-overview%}) is not declared in the markup like the other data-bound columns. It renders automatically when a `DetailTemplate` is added to the Grid. This built-in hierarchy expand/collapse column does not support being locked or other common configurations like managing its position, setting its width, or adding a title.
+
+To customize the hierarchy expand/collapse column, follow these steps:
+
+1. Hide the default hierarchy expand column using [custom CSS]({%slug themes-override%}).
+2. Add another [column with a template]({%slug grid-templates-column%}) for expanding and collapsing the detail templates through the [Grid state]({%slug components/grid/features/hierarchy%}#expand-rows-from-code).
+3. Disable the data operations for this column if you have enabled them in the Grid (filtering, sorting, etc.)
```

Comment:
This is necessary only if the column has a `Field`. Otherwise the data operations are disabled by default.